Fix unvisited JSON members

anothersimulacrum opened the pull request:
#### Summary
None

#### Purpose of change
Working towards re-enabling unvisited JSON member checking (see https://github.com/CleverRaven/Cataclysm-DDA/issues/64438).
Fix JSON errors.

#### Describe the solution
Enable unvisited member checking, game complains about these.

#### Additional context
Mods still need to be done.
Alerting the following people in case these data errors need other solutions
@dseguin @Venera3 https://github.com/CleverRaven/Cataclysm-DDA/commit/48b36faec137562960964170727325463a99c004
@MNG-cataclysm @LovamKicsiGazsii https://github.com/CleverRaven/Cataclysm-DDA/commit/a773ecec5e14e5d5419a68574ee70ebdf4af95a5 https://github.com/CleverRaven/Cataclysm-DDA/commit/70dc95c06db3e4659c1452609f8b25b362d15a39 https://github.com/CleverRaven/Cataclysm-DDA/commit/3aca0a2fd56ac5e6fc1c2a39d2247960b24edb14
@GuardianDll https://github.com/CleverRaven/Cataclysm-DDA/commit/8ad015a6af7073b1c0960bd46056cc690652ab6b https://github.com/CleverRaven/Cataclysm-DDA/commit/673a65cc15a0d10722ca0b0f5ac1e3b0a84467d8

anothersimulacrum committed:
Use "null", not JSON special null

null is a special value in JSON, but the parsing code expects a string
here to actually parse this value. Make it a string as expected.

anothersimulacrum committed:
Remove unused encumb_impacts_dodge

This was removed in 9d71fc88a376c4b70c93f5bdf0b83dab17834a74, but not
removed from the entries.

anothersimulacrum committed:
Remove unused description from construction

Constructions cannot use descriptions, and this is just a duplicate of
the description of the terrain, which is the description for the
construction anyways.

anothersimulacrum committed:
Remove sun_intensity

This was removed in 5525a3f38abde91df9de202de7d60d8bfce4db1b, but
5e94c7cc1d9deaf100c73afca09cee172378d97d added a weather containing it.
It was also not removed from the docs.

anothersimulacrum committed:
Remove 'repeat' from itemgroup

This appears to be confusion with the mapgen syntax.

anothersimulacrum committed:
Remove xedra_corpse container_item

It should have been container-item, but this whole thing was just an
accidental leftover from prototyping, so axe it.

anothersimulacrum committed:
Various minor typos

'attack_cost ' -> 'attack_cost'
flag -> flags
colo -> color
coutnt -> count
variants_type -> variant_type

anothersimulacrum committed:
Mutations don't have relic data

Co-authored-by: Anton Simakov <67688115+GuardianDll@users.noreply.github.com>

Looks like a simple misunderstanding, replace with enchantments.

anothersimulacrum committed:
Move data that should be in NPC classes from NPCs

NPC classes are the appropriate place for this, not NPC defs.

anothersimulacrum committed:
Variants don't support item attributes

Just remove them.

anothersimulacrum committed:
Move npc_lose_trait to effect

anothersimulacrum committed:
npc_lose_var doesn't take a value

anothersimulacrum committed:
No sealing here

anothersimulacrum committed:
Fix items using comestible properties that are not

Make one of them a comestible, remove the property in the other case

Venera3 reviewed (approved):
Dodge encumb got scored properly, some dumdum forgot to remove the field. Thanks!

GuardianDll commented on data/json/mutations/mutations.json:
```diff
@@ -7959,7 +7959,6 @@
     "points": 99,
     "valid": false,
     "description": "If you need to survive the deadliest bug attacks.",
```
```suggestion
    "description": "If you need to survive the deadliest bug attacks.",
    "enchantments": [ { "values": [ { "value": "MAX_HP", "add": 9000 } ] } ],
```

GuardianDll commented on data/json/mutations/mutations.json:
```diff
@@ -7969,7 +7968,6 @@
     "points": 99,
     "valid": false,
     "description": "Heal yourself from this stupid bug attacks.",
```
```suggestion
    "description": "Heal yourself from this stupid bug attacks.",
    "enchantments": [ { "values": [ { "value": "REGEN_HP", "add": 100 } ] } ],
```

anothersimulacrum commented:
It will work with no `to` field defined, but it expects a string for that field if it is defined. I'll remove the field.

FarkasKBence reviewed (approved):
At first I was confused why you removed the `repeat`s from the private resort, but then it dawned on me that distribution doesn't work like that.🤦

GuardianDll reviewed (approved):
Kinda sad the variants do not accept variety in stats, but thats okay

anothersimulacrum committed:
Remove null folded volume

Leaving it unset has the same effect, the part is not foldable. I'm not
sure why the reader (assign()) chokes on null here, though.

anothersimulacrum committed:
Remove members that don't belong

These aren't loaded by these objects. The container item for the pills
is already a bottle, and I'm not sure what density was aiming to
accomplish exactly.

anothersimulacrum committed:
Max range goes in spells, not spell monattacks

anothersimulacrum committed:
Typo fix
